File: Practica3/pila.py
class Pila:
    def __init__(self, n: int):
        self.arreglo: list[int] = [0] * n
        self.top: int = -1
        self.n: int = n

    def push(self, e: int):
        if self.isFull():
            print("Pila llena!")
        else:
            self.top += 1
            self.arreglo[self.top] = e

    def pop(self) -> int:
        if self.isEmpty():
            print("Pila vacía")
            return -1
        elemento: int = self.arreglo[self.top]
        self.top -= 1
        return elemento

    def peek(self) -> int:
        if self.isEmpty():
            print("Pila vacía")
            return -1
        return self.arreglo[self.top]

    def isEmpty(self) -> bool:
        return self.top == -1

    def isFull(self) -> bool:
        return self.top == self.n - 1

if __name__ == "__main__":
    pila = Pila(3)
    pila.push(10)
    pila.push(20)
    pila.push(30) 
    print(f"Elemento eliminado: {pila.pop()}")
    print(f"Elemento en la cima: {pila.peek()}")

File: Practica3/Pila/src/estructura/Pila.java
package estructura;

public class Pila {
    private long[] arreglo;
    private int top;
    private int n;

    public Pila(int n) {
        this.n = n;
        this.arreglo = new long[n];
        this.top = -1;
    }

    public void push(long e) {
        if (isFull()) {
            System.out.println("Pila llena");
            return;
        }
        arreglo[++top] = e;
    }

    public long pop() {
        if (isEmpty()) {
            System.out.println("Pila vacía");
            return -1;
        }
        return arreglo[top--];
    }

    public long peek() {
        if (isEmpty()) {
            System.out.println("Pila vacía");
            return -1;
        }
        return arreglo[top];
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public boolean isFull() {
        return top == n - 1;
    }

    public static void main(String[] args) {
        Pila pila = new Pila(6);

        System.out.println("Elemento eliminado: " + pila.pop());
        System.out.println("Elemento en la cima: " + pila.peek());
    }
}

